File: Easy/Equilibrium index of an array/equilibrium-index-of-an-array.java
//{ Driver Code Starts
import java.util.*;
class Index{
public static void main(String[] args)
 {
	 Scanner sc = new Scanner(System.in);
	 int T = sc.nextInt();
		while(T-- > 0)
		{
			int n = sc.nextInt();
			int[] a = new int[n];
			for(int i=0;i<n;i++)
			a[i] = sc.nextInt();
			GfG g = new GfG();
			System.out.println(g.findEquilibrium(a,n));
		}
}
}
// } Driver Code Ends


/*Complete the function below*/
class GfG
{
           public static int findEquilibrium(int arr[], int n)
              {
                int total_sum = 0;
                int check = 0;
                
                for(int i = 0; i<n; i++){
                    total_sum += arr[i];
                }
                
                for(int i = 0; i<n; i++){
                    total_sum -= arr[i];
                    if(check == total_sum){
                        return i;
                    }
                    check += arr[i];
                }
                
                return -1;
              }
}

File: Easy/Equilibrium index of an array/equilibrium-index-of-an-array.cpp
//{ Driver Code Starts
#include <iostream>
using namespace std;
#include <stdio.h>
 
int findEquilibrium(int [], int );
 
int main() {
	int T;
	cin>>T;
	while(T--)
	{
		int n;
		cin>>n;
		int a[n];
		for(int i=0;i<n;i++)
		cin>>a[i];
		cout<<findEquilibrium (a,n)<<endl;
	}
	// your code goes here
	return 0;
}
// } Driver Code Ends


/* You are required to complete this method*/
int findEquilibrium(int A[], int n)
{
        int total_sum = 0;
        int l_sum = 0;
        
        for(int i = 0; i<n; i++){
            total_sum += A[i];
        }
        
        for(int i = 0; i<n; i++){
            int r_sum = total_sum - l_sum - A[i];
            
            if(l_sum == r_sum){
                return i;
            }
            l_sum += A[i];
        }
        return -1;
}